src/Sylius/Component/Core/Model/ProductVariant.php

```diff
@@ -256,7 +256,15 @@ public function addChannelPricing(ChannelPricingInterface $channelPricing): void
     {
         if (!$this->hasChannelPricing($channelPricing)) {
             $channelPricing->setProductVariant($this);
-            $this->channelPricings->set($channelPricing->getChannelCode(), $channelPricing);
+            if($this->channelPricings->get($channelPricing->getChannelCode())){
```

Comment:
Can you describe what are you doing here?

Reply:
The code I have added prevents  persisting two channelPricings of the same channel so it will update the prices for the current one in case it already exists